warplib/maths/external_input_check_bool.cairo

```diff
@@ -0,0 +1,10 @@
+from starkware.cairo.common.math_cmp import is_le_felt
+
+func warp_external_input_check_bool{range_check_ptr}(x : felt):
+    let (inRange : felt) = is_le_felt(x, 1)
```

Comment:
Worth checking the step count to be sure, but I think `assert x * (x-1) = 0` would be more efficient

Reply:
Yeah, you are right. Changing out that assert statement brings the step count down from 46->11.